### ISSUE TITLE
System.Net.Http, Version=4.2.0.0 returns System.IO.FileNotFoundException

### ISSUE THREAD

**Issue by alessandroros** [davetimmins/Anywhere.ArcGIS#14]
Hi @davetimmins, first of all thanks a lot for this project, I think it's a really useful package for everyone's working with ArcGIS Rest Services.
I installed Anywhere.ArcGIS version 1.2,1 on  my project which is developed with .NET Framework 4.6.2.
On compilation everything's fine, but while running it throws an error like this:

> Could not load file or assembly 'System.Net.Http, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file.
System.IO.FileNotFoundException"

Have you ever experienced this issue? I saw you're targeting the package for netstandard 2.0, so .NET 4.6.2 should be fine, right?

I think it should be an issue of Microsoft and Visual Studio: 
https://github.com/dotnet/corefx/issues/23306

But I wanted to know if someone is having this issue or it's just me. 
Anyway thanks for your work!

**Comment by davetimmins** [comment]
Hi, I just created a new .NET framework console app targetting v4.6.2, installed Anywhere.ArcGIS from NuGet and ran the Query sample code on the readme and it worked fine. There's nothing in the app.config for binding redirects but if you have existing ones in your config then that could be something to look at. The reference shows as
![image](https://user-images.githubusercontent.com/1118347/35237029-4120ae48-000e-11e8-985a-fe636956258d.png)

I'm using this lib in a couple of projects that target .NET 4.6.2 too and haven't come across the problem in those. Where are you running the code when you get the error, is it on your local machine or on a (cloud) server?

**Comment by alessandroros** [comment]
Ok thanks for the advice on the app.config. I'll check today and let you know. 
Just to reproduce the same environment which VS version are you using? I'm using the latest 15.5.4 (VS Community 2017). I'm also using Autofac for DI but this shouldn't be an issue with it.

**Comment by davetimmins** [comment]
Yea I'm using that version too. If you can send me a zip of the solution that has the problem I can try running it to see if I can reproduce it

**Comment by alessandroros** [comment]
Ok I just finished to create a clean project with a service which calls your library.
What I forgot to say is that I'm using other packages like owin, autofac, swashbuckle and automapper in my project. In the uploaded project I installed all these packages i'm using and then I was able to reproduce the issue. 

[TestForDaveTimmins2Cleaned.zip](https://github.com/davetimmins/Anywhere.ArcGIS/files/1655449/TestForDaveTimmins2Cleaned.zip)


I split the solution in 2 projects:

- The WebApi project is using System.Net.Http 4.0.0.0
![image](https://user-images.githubusercontent.com/3214597/35271200-9dda8816-0031-11e8-90c9-e32236aa5e46.png).

- The Service project is using System.Net.Http 4.2.0.0 since it's a reference imported from the package Anywhere.ArcGIS.

You can get the error by running the solution and doing the following request *the only service available throws the error):

![image](https://user-images.githubusercontent.com/3214597/35271420-5b46165e-0032-11e8-9c4f-9cda73a0e714.png)

In any case thanks for your time and for the work you're doing! :1st_place_medal:

**Comment by davetimmins** [comment]
OK I managed to get it working. To fix the error above I added a binding redirect to the web.config

```
<dependentAssembly>
  <assemblyIdentity name="System.Net.Http" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
  <bindingRedirect oldVersion="0.0.0.0-4.2.0.0" newVersion="4.0.0.0" />
</dependentAssembly>
```

That then told me that there was a version mismatch for Json.NET so I updated that to version 10.0.3 in the WebApi project which made the code run but not return a value to the UI. This was due to a threading issue so lastly I changed the async call to 
`var result = await _gateway.Query<Point>(query).ConfigureAwait(false);` 
and that runs correctly now.
